Source/WebKit/UIProcess/API/gtk/PageClientImpl.cpp

```diff
@@ -552,13 +554,19 @@ void PageClientImpl::derefView()
 void PageClientImpl::requestDOMPasteAccess(WebCore::DOMPasteAccessCategory, const IntRect&, const String& originIdentifier, CompletionHandler<void(WebCore::DOMPasteAccessResponse)>&& completionHandler)
 {
     auto& clipboard = Clipboard::get("CLIPBOARD"_s);
-    clipboard.readBuffer(PasteboardCustomData::gtkType().characters(), [originIdentifier, completionHandler = WTFMove(completionHandler)](Ref<SharedBuffer>&& buffer) mutable {
+    clipboard.readBuffer(PasteboardCustomData::gtkType().characters(), [weakWebView = GWeakPtr<GtkWidget>(m_viewWidget), originIdentifier, completionHandler = WTFMove(completionHandler)](Ref<SharedBuffer>&& buffer) mutable {
         if (PasteboardCustomData::fromSharedBuffer(buffer.get()).origin() == originIdentifier) {
             completionHandler(DOMPasteAccessResponse::GrantedForGesture);
             return;
         }
 
-        completionHandler(DOMPasteAccessResponse::DeniedForGesture);
+        if (!WEBKIT_IS_WEB_VIEW(weakWebView.get())) {
```

Comment:
Shouldn't this be:

```suggestion
        if (!weakWebView.get()) {
```

? It's either a WebKitWebView or it's nullptr, right?

Reply:
No, it could be WebKitWebViewBase which doesn't support permission requests.